extensions/amp-vimeo/0.1/amp-vimeo.js

```diff
@@ -13,66 +13,371 @@
+  return (/^(https?:)?\/\/((player|www).)?vimeo.com(?=$|\/)/).test(url);
+}
+
+
+/**
+ * @param {?} anything
+ * @return {boolean}
+ */
+function isJsonOrObj(anything) {
+  return anything && (
+    isObject(anything) || startsWith(/** @type {string} */ (anything), '{'));
+}
+
+
+/** @private {!Object<string, string>} */
+// If the item does not have a value, the event will not be forwarded
+// automatically, but it will be listened to.
+const VIMEO_EVENTS = {
+  'play': VideoEvents.PLAYING,
+  'pause': VideoEvents.PAUSE,
+  'ended': VideoEvents.ENDED,
+  'volumechange': null,
+};
+
+
+/** @implements {../../../src/video-interface.VideoInterface} */
 class AmpVimeo extends AMP.BaseElement {
 
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+
     /** @private {?Element} */
     this.iframe_ = null;
+
+    /** @private {function():string} */
+    this.setVolumeMethod_ = once(() => getMethodName('volume', 'set'));
+
+    /** @private {boolean} */
+    this.muted_ = false;
+
+    /**
+     * @param {!Event} e
+     * @private
+     */
+    this.boundOnMessage_ = e => this.onMessage_(e);
+
+    /** @private {!UnlistenDef|null} */
+    this.unlistenFrame_ = null;
   }
 
   /**
    * @param {boolean=} onLayout
    * @override
    */
   preconnectCallback(onLayout) {
-    this.preconnect.url('https://player.vimeo.com', onLayout);
+    const {preconnect} = this;
+    preconnect.url('https://player.vimeo.com', onLayout);
     // Host that Vimeo uses to serve poster frames needed by player.
-    this.preconnect.url('https://i.vimeocdn.com', onLayout);
+    preconnect.url('https://i.vimeocdn.com', onLayout);
     // Host that Vimeo uses to serve JS, CSS and other assets needed.
-    this.preconnect.url('https://f.vimeocdn.com', onLayout);
+    preconnect.url('https://f.vimeocdn.com', onLayout);
   }
 
-  /** @override */
+  /**
+   * @override
+   * @inheritdoc
+   */
   isLayoutSupported(layout) {
     return isLayoutSizeDefined(layout);
   }
 
+  /** @override */
+  buildCallback() {
+    installVideoManagerForDoc(this.getAmpDoc());
+  }
+
   /** @override */
   layoutCallback() {
-    const videoid = user().assert(
-        this.element.getAttribute('data-videoid'),
+    const {element} = this;
+    const vidId = user().assert(
+        element.getAttribute('data-videoid'),
         'The data-videoid attribute is required for <amp-vimeo> %s',
-        this.element);
+        element);
+
     // See
     // https://developer.vimeo.com/player/embedding
-    const iframe = this.element.ownerDocument.createElement('iframe');
-    iframe.setAttribute('frameborder', '0');
-    iframe.setAttribute('allowfullscreen', 'true');
-    iframe.src = 'https://player.vimeo.com/video/' + encodeURIComponent(
-        videoid);
+    const iframe =
+        htmlFor(element)`<iframe frameborder=0 allowfullscreen></iframe>`;
+
+    let src = `https://player.vimeo.com/video/${encodeURIComponent(vidId)}`;
+
+    if (this.hasAutoplay_()) {
```

Comment:
as discussed, this would to use the isFullscreenSupported && hasFullscreen which unfortunately is async putting all of this a frame out but that should be fine.

Reply:
I'm guessing you meant `isAutoplaySupported`? :) if so, then fixed.

Reply:
Lol, context switch overflow